lib/matplotlib/image.py

```diff
@@ -443,73 +448,35 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
-                # we are done with the mask, delete from namespace to be sure!
+                out_alpha = _resample(self, mask, out_shape, t, resample=True)
+                # done with the mask now, delete from namespace to be sure!
                 del mask
-                # Agg updates the out_mask in place.  If the pixel has
-                # no image data it will not be updated (and still be 0
-                # as we initialized it), if input data that would go
-                # into that output pixel than it will be `nan`, if all
-                # the input data for a pixel is good it will be 1, and
-                # if there is _some_ good data in that output pixel it
-                # will be between [0, 1] (such as a rotated image).
-
-                out_alpha = np.array(out_mask)
-                out_mask = np.isnan(out_mask)
+                # Agg updates out_alpha in place.  If the pixel has no image
+                # data it will not be updated (and still be 0 as we initialized
+                # it), if input data that would go into that output pixel than
+                # it will be `nan`, if all the input data for a pixel is good
+                # it will be 1, and if there is _some_ good data in that output
+                # pixel it will be between [0, 1] (such as a rotated image).
+                out_mask = np.isnan(out_alpha)
                 out_alpha[out_mask] = 1
-
                 # mask and run through the norm
                 output = self.norm(np.ma.masked_array(A_resampled, out_mask))
             else:
-                # Always convert to RGBA, even if only RGB input
                 if A.shape[2] == 3:
                     A = _rgb_to_rgba(A)
-                elif A.shape[2] != 4:
-                    raise ValueError("Invalid shape {} for image data"
-                                     .format(A.shape))
-
-                output = np.zeros((out_height, out_width, 4), dtype=A.dtype)
-                output_a = np.zeros((out_height, out_width), dtype=A.dtype)
-
                 alpha = self.get_alpha()
                 if alpha is None:
                     alpha = 1
-
-                #resample alpha channel
-                alpha_channel = A[..., 3]
-                _image.resample(
-                    alpha_channel, output_a, t,
-                    _interpd_[self.get_interpolation()],
-                    self.get_resample(), alpha,
-                    self.get_filternorm(), self.get_filterrad())
-
-                #resample rgb channels
-                A = _rgb_to_rgba(A[..., :3])
-                _image.resample(
-                    A, output, t,
-                    _interpd_[self.get_interpolation()],
-                    self.get_resample(), alpha,
-                    self.get_filternorm(), self.get_filterrad())
-
-                #recombine rgb and alpha channels
-                output[..., 3] = output_a
+                output_a = _resample(  # resample alpha channel
```

Comment:
even though it's in the original code, I would rename it to make it more clear.
```suggestion
                output_alpha = _resample(  # resample alpha channel
```

Reply:
sure